Unable to generate a service account key with a specific TTL

I'd like the ability to generate Service Account key with a specific TTL (not the one by default or the one configured in gcp/config).

Following the [documentation](https://www.vaultproject.io/api/secret/gcp/index.html#generate-secret-iam-service-account-creds-service-account-key), I tried to execute:

`curl --header "X-Vault-Token: XXXX" --request POST --data @payload.json 'https://<VAULT_ADDR>/v1/gcp/key/<ROLE_NAME>'`

With this payload:
```
{
  "roleset": "<ROLE_NAME>",
  "key_algorithm": "TYPE_GOOGLE_CREDENTIALS_FILE",
  "key_type": "KEY_ALG_RSA_2048",
  "ttl": "5m"
}
```

Even though the "ttl" is not listed as available parameter in the documentation, it looks like [it can be specified when looking at the code](https://github.com/hashicorp/vault-plugin-secrets-gcp/blob/master/plugin/secrets_service_account_key.go#L61-L64).

Anyways, I'm getting always the same result with or without the "ttl" parameter:

> {"errors":["1 error occurred:\n\t* internal error\n\n"]}

and this error appears in vault logs:

`[ERROR] core: failed to run existence check: error="roleset name is required"`

Looking at the code, it looks like the declared field name is `roleset` [inside the Fields schema](https://github.com/hashicorp/vault-plugin-secrets-gcp/blob/master/plugin/secrets_service_account_key.go#L47), while a field `name` is expected inside the corresponding [ExistenceCheck](https://github.com/hashicorp/vault-plugin-secrets-gcp/blob/556b7c55d6db11a34eef1bbf6bc455bedfecdaf6/plugin/path_role_set.go#L124-L127).

--- a/plugin/path_role_set.go
+++ b/plugin/path_role_set.go
@@ -45,7 +45,7 @@ func pathRoleSet(b *backend) *framework.Path {
 				Description: `List of OAuth scopes to assign to credentials generated under this role set`,
 			},
 		},
-		ExistenceCheck: b.pathRoleSetExistenceCheck,
+		ExistenceCheck: b.pathRoleSetExistenceCheck("name"),
 		Operations: map[logical.Operation]framework.OperationHandler{
 			logical.DeleteOperation: &framework.PathOperation{
 				Callback: b.pathRoleSetDelete,
@@ -89,7 +89,7 @@ func pathRoleSetRotateAccount(b *backend) *framework.Path {
 				Description: "Name of the role.",
 			},
 		},
-		ExistenceCheck: b.pathRoleSetExistenceCheck,
+		ExistenceCheck: b.pathRoleSetExistenceCheck("name"),
 		Operations: map[logical.Operation]framework.OperationHandler{
 			logical.UpdateOperation: &framework.PathOperation{
 				Callback: b.pathRoleSetRotateAccount,
@@ -109,7 +109,7 @@ func pathRoleSetRotateKey(b *backend) *framework.Path {
 				Description: "Name of the role.",
 			},
 		},
-		ExistenceCheck: b.pathRoleSetExistenceCheck,
+		ExistenceCheck: b.pathRoleSetExistenceCheck("name"),
 		Operations: map[logical.Operation]framework.OperationHandler{
 			logical.UpdateOperation: &framework.PathOperation{
 				Callback: b.pathRoleSetRotateKey,
@@ -120,18 +120,21 @@ func pathRoleSetRotateKey(b *backend) *framework.Path {
 	}
 }
 
-func (b *backend) pathRoleSetExistenceCheck(ctx context.Context, req *logical.Request, d *framework.FieldData) (bool, error) {
-	nameRaw, ok := d.GetOk("name")
-	if !ok {
-		return false, errors.New("roleset name is required")
-	}
+func (b *backend) pathRoleSetExistenceCheck(rolesetFieldName string) framework.ExistenceFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (bool, error) {
+		// check for either name or roleset
+		nameRaw, ok := d.GetOk(rolesetFieldName)
+		if !ok {
+			return false, errors.New("roleset name is required")
+		}
 
-	rs, err := getRoleSet(nameRaw.(string), ctx, req.Storage)
-	if err != nil {
-		return false, err
-	}
+		rs, err := getRoleSet(nameRaw.(string), ctx, req.Storage)
+		if err != nil {
+			return false, err
+		}
 
-	return rs != nil, nil
+		return rs != nil, nil
+	}
 }
 
 func (b *backend) pathRoleSetRead(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {

--- a/plugin/secrets_access_token.go
+++ b/plugin/secrets_access_token.go
@@ -22,7 +22,7 @@ func pathSecretAccessToken(b *backend) *framework.Path {
 				Description: "Required. Name of the role set.",
 			},
 		},
-		ExistenceCheck: b.pathRoleSetExistenceCheck,
+		ExistenceCheck: b.pathRoleSetExistenceCheck("roleset"),
 		Operations: map[logical.Operation]framework.OperationHandler{
 			logical.ReadOperation:   &framework.PathOperation{Callback: b.pathAccessToken},
 			logical.UpdateOperation: &framework.PathOperation{Callback: b.pathAccessToken},

--- a/plugin/secrets_service_account_key.go
+++ b/plugin/secrets_service_account_key.go
@@ -63,7 +63,7 @@ func pathSecretServiceAccountKey(b *backend) *framework.Path {
 				Description: "Lifetime of the service account key",
 			},
 		},
-		ExistenceCheck: b.pathRoleSetExistenceCheck,
+		ExistenceCheck: b.pathRoleSetExistenceCheck("roleset"),
 		Operations: map[logical.Operation]framework.OperationHandler{
 			logical.ReadOperation:   &framework.PathOperation{Callback: b.pathServiceAccountKey},
 			logical.UpdateOperation: &framework.PathOperation{Callback: b.pathServiceAccountKey},

